service_configuration_lib/spark_config.py

```diff
@@ -994,9 +1001,80 @@ def _get_valid_jira_ticket(self, jira_ticket: Optional[str]) -> Optional[str]:
+
+        Returns:
+            The validated Jira ticket string if valid, otherwise None.
+
+        Args:
+            cluster_manager: The cluster manager being used
+            user: The user running the job
+            jira_ticket: The Jira ticket provided by the user
+        """
+        flag_enabled = self.mandatory_default_spark_srv_conf.get('spark.yelp.jira_ticket.enabled', 'false')
+        valid_ticket = self._get_valid_jira_ticket(jira_ticket)
+
+        # Skip validation for local cluster manager or exempt users
+        if cluster_manager == 'local' or user in TICKET_NOT_REQUIRED_USERS:
+            log.debug('Jira ticket check not required for this job configuration.')
+            # If exempt, we still pass through the original ticket if it's valid,
+            # otherwise None. This allows exempt users like tron to still have their valid tickets
+            # (if provided) attached as labels, without forcing validation.
+            return valid_ticket
+
+        if valid_ticket is None:
+            if flag_enabled == 'true':
+                error_msg = (
+                    f'Job requires a valid Jira ticket (format PROJ-1234).\n'
+                    f'Jira ticket check is enabled, but ticket "{jira_ticket}" is '
+                    f'missing or invalid for user "{user}".\n'
+                    'Please pass the parameter as: paasta spark-run --jira-ticket=PROJ-1234 \n'
+                    'For more information: http://y/spark-jira-ticket-param \n'
+                    'If you have questions, please reach out to #spark on Slack.\n'
+                )
+                raise RuntimeError(error_msg)
+            else:
+                warning_message = (
+                    f'Jira ticket check is configured, but ticket is missing or invalid for user "{user}". '
+                    f'Proceeding with job execution. Original ticket value: "{jira_ticket}". '
+                    'Please pass the parameter as: paasta spark-run --jira-ticket=PROJ-1234 '
+                    'For more information: http://y/spark-jira-ticket-param '
+                    'If you have questions, please reach out to #spark on Slack. '
+                )
+                if clog:
+                    try:
+                        clog.config.configure(
+                            scribe_host='169.254.255.254',  # Standard Scribe host
+                            scribe_port='1463',             # Standard Scribe port
+                            monk_disable=False,           # Ensure Monk (for clog) is enabled
+                            scribe_disable=False,         # Ensure Scribe is enabled
+                        )
+                        log_payload = {
+                            'timestamp': int(time.time()),
+                            'event': 'jira_ticket_validation_warning',
+                            'level': 'WARNING',
+                            'reason': 'Ticket missing or invalid. See http://y/spark-jira-ticket-param',
```

Comment:
nit:
```suggestion
                            'reason': 'Ticket missing or invalid.
```
we (well, y'all) will be the only folks looking at this stream - so we can save a couple bytes and leave out the user-facing docs